File: 456_132_pattern/pattern.java
class Solution {
    public boolean find132pattern(int[] nums) {
        int i, j, k, n = nums.length;
        if (n < 3) {
            return false;
        }
        i = 0;
        for (j=1; j < n -1 ; j++) {
            if (nums[j] < nums[i]) {
                i = j;
            } else {
                for (k = j+1; k < n; k++) {
                    if (nums[j] > nums[k] && nums[k] > nums[i]) {
                        return true;
                    }
                }
            }
        }
        return false;
    }
}

File: 456_132_pattern/pattern.py
from typing import List


class Solution:
    def find132pattern(self, nums: List[int]) -> bool:
        if len(nums) < 3:
            return False
        i = 0
        for j in range(1, len(nums) - 1):
            if nums[j] > nums[i]:
                # i 和 j已经找到 找k
                for k in range(j + 1, len(nums)):
                    if nums[j] > nums[k] and nums[k] > nums[i]:
                        return True
            else:
                i = j  # 第一个数取最小值
        return False
